src/pytest_plugins/filler/witness.py

```diff
@@ -0,0 +1,119 @@
+"""
+Pytest plugin for witness functionality.
+
+Provides --witness command-line option that installs the witness-filler tool
+and generates execution witness data for blockchain test fixtures when enabled.
+"""
+
+import subprocess
+from typing import Any, Callable
+
+import pytest
+
+from ethereum_test_fixtures.blockchain import FixtureBlock, WitnessChunk
+
+
+def pytest_addoption(parser: pytest.Parser):
+    """Add witness command-line options to pytest."""
+    witness_group = parser.getgroup("witness", "Arguments for witness functionality")
+    witness_group.addoption(
+        "--witness",
+        "--witness-the-fitness",
+        action="store_true",
+        dest="witness",
+        default=False,
+        help=(
+            "Install the witness-filler tool and generate execution witness data for blockchain "
+            "test fixtures."
+        ),
+    )
+
+
+def pytest_configure(config):
+    """
+    Pytest hook called after command line options have been parsed.
+
+    If --witness is enabled, installs the witness-filler tool from the specified
+    git repository.
+    """
+    if config.getoption("witness"):
+        print("🔧 Installing witness-filler tool from kevaundray/reth...")
+        print("   This may take several minutes for first-time compilation...")
+
+        result = subprocess.run(
+            [
+                "cargo",
+                "install",
+                "--git",
+                "https://github.com/kevaundray/reth.git",
+                "--rev",
+                "8016a8a5736e4427b3d285c82cd39c4ece70f8c4",
+                "witness-filler",
+            ],
+        )
```

Comment:
This doesn't make any sense to me, this is a big side-effect of running this command, it should be a requirement rather than an automatic install.

Reply:
> We won't be running with `--witness` typically. We can always revisit later.

Famous last words before CI breaks because we forgot and tried to install something as non-root.

Also, what happens when users start to deviate from this? Can we guarantee that zkevm and other stateless efforts are never going to use different versions?

Also, if this external tool needs update at any point, it now requires a PR to EEST.

Reply:
> I think we can point to a branch instead of a commit maybe? --branch branch-name instead of --rev <commit>?
It's a similar problem IMO. `witness-filler` has to be already in path for this to work, we could at most check that it's indeed there and fail if not.

The CI workflow (if any) is the one responsible to install the correct version before running the `fill` command.

Reply:
Ok, ok, fair enough, I know it's not very clean. Let me take it out.